tests/unit/components/tumblr-post-test.js

```diff
@@ -1,20 +1,52 @@
+import Ember from 'ember';
 import { moduleForComponent, test } from 'ember-qunit';
 
 moduleForComponent('tumblr-post', 'Unit | Component | tumblr post', {
-  // Specify the other units that are required for this test
-  // needs: ['component:foo', 'helper:bar'],
   needs: ['component:tumblr-post-text'],
   unit: true
 });
 
-test('it renders', function(assert) {
+test('it renders', function (assert) {
   assert.expect(2);
 
   // Creates the component instance
-  var component = this.subject();
+  const component = this.subject();
   assert.equal(component._state, 'preRender');
 
   // Renders the component to the page
   this.render();
   assert.equal(component._state, 'inDOM');
 });
+
+test('defaults', function (assert) {
+  const component = this.subject();
+
+  assert.ok(!Ember.isPresent(component.get('postsRoute')), 'postsRoute is not defined by default');
+  assert.ok(!component.get('collapsible'), 'post is not collapsible by default');
+  assert.ok(component.get('collapsed'), 'post is set to collapse by default if collapsible');
+  assert.ok(!component.get('isCollapsed'), 'post will not collapse by default, because it is not collapsible');
+});
+
+test('isCollapsed', function (assert) {
+  const component = this.subject();
+
+  component.set('collapsible', true);
+  assert.ok(component.get('isCollapsed'), 'post is collapsed by default when collapsible');
+
+  component.set('collapsed', false);
+  assert.ok(!component.get('isCollapsed'), 'post is not collapsed');
+});
+
+test('expandButtonText', function (assert) {
+  const collapsedText = 'foo';
+  const expandedText = 'bar';
+  const component = this.subject({
```

Comment:
'const' is available in ES6 (use esnext option) or Mozilla JS extensions (use moz).